libcloud/common/exception.py

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class BaseHTTPException(Exception):
+
+    """
+    The base exception class for all exception raises.
+    """
+
+    def __init__(self, code, message, headers=None):
+
+        self.message = message
+        self.code = code
+        self.headers = headers
+
```

Comment:
Did you verify that treating the exception as a string returns self.message?  I would assume you'd need an ```def __str__(self)``` method, but maybe self.message is the default?